.github/workflows/ion-java-performance-regression-detector.yml

```diff
@@ -46,22 +46,42 @@ jobs:
           name: ${{env.test_data_id}}
           path: testData
 
+  select-runner-type:
+    # If you want to use codebuild runners for your personal fork, follow the instructions to set
+    # up a codebuild project. https://docs.aws.amazon.com/codebuild/latest/userguide/action-runner.html
+    # Then, create a repository variable for your fork named `CODEBUILD_PROJECT_NAME` with the name
+    # of the project you created.
+    name: Select Runner Type
+    runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on internal ones as push automatically builds
```

Comment:
Can you help me understand this better? Where exactly do we *not* want to use CodeBuild, and why? And we fall back to `ubuntu-latest` when CodeBuild is disabled?

Reply:
I thought that this would be enough to explain. Feel free to suggest additions.
> If you want to use codebuild runners for your personal fork, follow the instructions to set
up a codebuild project. https://docs.aws.amazon.com/codebuild/latest/userguide/action-runner.html
Then, create a repository variable for your fork named `CODEBUILD_PROJECT_NAME` with the name
of the project you created.

Basically, if you want the perf workflow to run when you push to your fork of the repo, either you need to have codebuild runners setup (which will cost money), or we need to have a fallback option.

I suppose, though, that we might decide that any results on a different runner type are too different to be even the slightest bit valid. In that case, we might rewrite this workflow so that it is entirely skipped if there are no codebuild runners.

Reply:
I think part of my confusion is around the fact that I don't know the intended definition of "external" and "internal" PRs. Looking more closely, it looks like "internal" PRs are ones submitted directly against `amazon-ion/ion-java`, while "external" PRs covers everything else. And so this block is here just to help, e.g., users with forks still be able to run the perf workflow, falling back to Ubuntu if they don't have CodeBuild. Is that right?

Reply:
Oh, the "internal"/"external" PRs is just to prevent duplicate workflows, and it's something that is already used in our other workflows. When you create a PR from a fork repo to a source repo and a workflow has both `push` and `pull_request` triggers, it kicks off the Github workflow for both the push (GitHub pushes PR commits to the target repo somehow) and the actual pull request. The workaround is actually on L58.